src/roman_datamodels/datamodels.py

```diff
@@ -318,10 +323,180 @@ class RampFitOutputModel(DataModel):
+        #                  iscopy=False, **kwargs):
+        super().__init__(init=None, **kwargs)
+
+        self._models = []
+        self._iscopy = iscopy
+        # self.asn_exptypes = asn_exptypes
+        # self.asn_n_members = asn_n_members
+        # self.asn_table = {}
+        # self.asn_table_name = None
+        # self.asn_pool_name = None
+
+        self._memmap = kwargs.get("memmap", False)
+        self._return_open = kwargs.get('return_open', True)
+        self._save_open = kwargs.get('save_open', True)
+
+        if asn_file_path:
+            with builtins.open(asn_file_path, 'r') as asn_file:
+                asn_schema = json.load(asn_file)
+
+            for product in asn_schema['products']:
+                for member in product['members']:
+                    if self._iscopy:
+                        member_model = open(model_file_path + member['expname'])
+                    else:
+                        member_model = "null"
+
+                    self._models.append(
+                        # {
+                        #     "name" : member['expname'],
+                        #     "datamodel" : member_model
+                        # }
+                        member_model
+                    )
+        elif isinstance(init, list):
+            # only append list items to self._models if all items are either 
+            # strings (i.e. path to an ASDF file) or instances of DataModel
+
+            is_all_string = all(isinstance(x, str) for x in init)
+            is_all_roman_datamodels = all(isinstance(x, DataModel) for x in init)
+
+            if is_all_string or is_all_roman_datamodels:
+                self._models.extend(init)
+            else:
+                raise TypeError(
+                        "Input must be a list of strings (full path to ASDF files) or Roman datamodels.")
+        else:
+            raise TypeError(
+                        "Input must be an ASN filepath or list of either strings (full path to ASDF files) or Roman datamodels.")
+
+    def __len__(self):
+        return len(self._models)
+
+    def __getitem__(self, index):
+        m = self._models[index]
+        if not isinstance(m, DataModel) and self._return_open:
+            m = open(m, memmap=self._memmap)
+        return m
+    
+    def __setitem__(self, index, model):
+        self._models[index] = model
+
+    def __delitem__(self, index):
+        del self._models[index]
+
+    def __iter__(self):
+        for model in self._models:
+            if not isinstance(model, DataModel) and self._return_open:
+                model = open(model, memmap=self._memmap)
+            yield model
+
+    def insert(self, index, model):
+        self._models.insert(index, model)
+
+    def append(self, model):
+        self._models.append(model)
+
+    def extend(self, model):
+        self._models.extend(model)
+
+    def pop(self, index=-1):
+        self._models.pop(index)
+
+    def copy(self, memo=None):
```

Comment:
Is this method needed now? I would remove it if not. I still don't understand when we need to make a copy of a `ModelContainer`.

Reply:
No, it's not needed anymore.